ifenslave: Add ifenslave package

terasawa-lin opened the pull request:
Modifications to the recipe
add:
DEBIAN_PATCH_TYPE
DEBIAN_UNPACK_DIR

remove:
SRC_URI
S

# Purpose of pull request

This pr adds xinetd package based on meta-openembedded/meta-networking/recipes-support/ifenslave/ifenslave_2.9.bb

# Test
## How to install
Add the following to conf/local.conf.
```
IMAGE_INSTALL_append = " ifenslave"
```
CONFIG_BONDING must be enabled
Set as follows.
```
$ bitbake linux-base -c menuconfig
-> Device Drivers
 -> Network device support
 [*] Network core driver support
 <M>   Bonding driver support
```
## How to test

Edit /etc/network/interfaces, comment out the default eth0 and eth1 descriptions, and add:
```
# vi /etc/network/interfaces

#Wired or wireless interfaces
#auto eth0
#iface eth0 inet dhcp
#iface eth1 inet dhcp

#Define slaves
auto eth0
iface eth0 inet manual
bond-master bond0
bond-primary eth0
bond-mode active-backup

auto eth1
iface eth1 inet manual
bond-master bond0
bond-primary eth0
bond-mode active-backup

#Define master
auto bond0
iface bond0 inet dhcp
bond-slaves none
bond-primary eth0
bond-mode active-backup
bond-miimon 100
```

Create a /etc/modprobe.d/bonding.conf file and write it as follows:
```
# vi /etc/modprobe.d/bonding.conf
alias bond0 bonding
```

Run following command from console
```
# modprobe bonding
# reboot
```

## Test result
After connecting the LAN cable to eth0 and eth1, execute the ping command to confirm that communication is possible even if the LAN cable is connected and disconnected during execution.
```
# ping 8.8.8.8
```

hiraku-wfs commented on recipes-debian/ifenslave/ifenslave_debian.bb:
```diff
@@ -0,0 +1,35 @@
+inherit manpages
+MAN_PKG = "${PN}"
+
+SRCREV = "42bfbb9beb924672ca86b86e9679ac3d6b87d992"
```
Is this used?

terasawa-lin replied:
Let me confirm

terasawa-lin replied:
I confirmed that there was no problem, so I deleted it.

masami256 commented:
Tested on qemuarm64.

Run qemu by following command.

```
sudo /home/masami/emlinux/qemuarm64-build/tmp-glibc/work/x86_64-linux/qemu-helper-native/1.0-r1/recipe-sysroot-native/usr/bin/qemu-system-aarch64 \
-device virtio-net-device,netdev=net0,mac=52:54:00:12:34:02 -netdev user,id=net0,net=10.0.2.0/24,dhcpstart=10.0.2.15 \
-device virtio-net-device,netdev=net1,mac=52:54:00:12:34:03 -netdev user,id=net1,net=10.0.2.0/24,dhcpstart=10.0.2.15 \
-drive id=disk0,file=/home/masami/emlinux/qemuarm64-build/tmp-glibc/deploy/images/qemuarm64/core-image-minimal-qemuarm64-20210312071445.rootfs.ext4,if=none,format=raw -device virtio-blk-device,drive=disk0 \
-show-cursor -device VGA,edid=on -device qemu-xhci -device usb-tablet -device usb-kbd -object rng-random,filename=/dev/urandom,id=rng0 -device virtio-rng-pci,rng=rng0  \
-nographic -machine virt -cpu cortex-a57 -m 512 -serial mon:stdio -serial null \
-kernel /home/masami/emlinux/qemuarm64-build/tmp-glibc/deploy/images/qemuarm64/Image--git0+9b31f372ed-r0-qemuarm64-20210312060615.bin \
-append 'root=/dev/vda rw highres=off  console=ttyS0 mem=512M  console=ttyAMA0 ' 
```

While ping to 10.0.2.2,  link down/up eth0 didn't affect ping process.

terasawa-lin committed:
ifenslave: Add ifenslave package

Modifications to the recipe
add:
DEBIAN_PATCH_TYPE
DEBIAN_UNPACK_DIR

remove:
SRC_URI
S

hiraku-wfs commented:
Thanks!
